nixos/modules/system/activation/activatable-system.nix

```diff
@@ -22,28 +22,30 @@ let
     chmod u+x $out/activate $out/dry-activate
     unset activationScript dryActivationScript
 
-    mkdir $out/bin
-    substitute ${./switch-to-configuration.pl} $out/bin/switch-to-configuration \
-      --subst-var out \
-      --subst-var-by toplevel ''${!toplevelVar} \
-      --subst-var-by coreutils "${pkgs.coreutils}" \
-      --subst-var-by distroId ${lib.escapeShellArg config.system.nixos.distroId} \
-      --subst-var-by installBootLoader ${lib.escapeShellArg config.system.build.installBootLoader} \
-      --subst-var-by localeArchive "${config.i18n.glibcLocales}/lib/locale/locale-archive" \
-      --subst-var-by perl "${perlWrapped}" \
-      --subst-var-by shell "${pkgs.bash}/bin/sh" \
-      --subst-var-by su "${pkgs.shadow.su}/bin/su" \
-      --subst-var-by systemd "${config.systemd.package}" \
-      --subst-var-by utillinux "${pkgs.util-linux}" \
-      ;
+    ${lib.optionalString config.system.rebuildable ''
```

Comment:
Could you move this to a new file, `switchable-system.nix`?

Reply:
We would need a new variable (e.g. `system.build.systemBuilderCommands`) to communicate between the new file and the currently local `systemBuilderCommands` in activatable-system though. Is this desired?

Reply:
We already have `system.systemBuilderCommands`, with merging, so I think you could add to that without introducing any new options.

Reply:
Sadly that breaks the separateActivationScript. So we have to introduce a new variable or keep the changes in activatable-system

Reply:
`separateActivationScript` is a fairly new feature, and its name does not suggest switching.
I think it'd be ok not to support that, and perhaps provide a `separateSwitchScript` in addition.

- `separateActivationScript` is suitable for building a pre-activated immutable image without leaving the activation script around.
- `separateSwitchScript` is suitable for switching and then garbage collecting the switching tools, like some tools, or perhaps even perl, which interprets the switch script.

In such a case, `separateSwitchScript` could call `separateActivationScript`, or maybe override it to add its own binary. It wouldn't need to send its guts to the activation script module anymore.

You could implement `separateSwitchScript`, or make `system.activatableSystemBuilderCommands` an `internal` option and document that it may go away, referring to this discussion.

Reply:
I went with adding a comment to `system.activatableSystemBuilderCommands`. The option should already be internal because I just override the description of `options.system.systemBuilderCommands`. 

A `separateSwitchScript` might be hard to wire up because of the way `types.unspecified` merges definitions. We need to set `system.systemBuilderArgs.toplevelVar = "out";` in switchable-system when `system.activatable = false` for for switch-to-configuration to even *build* correctly. However we can only set this once. If I also set `toplevelVar` in switchable-system we get `toplevelVar = "outout" because it merges the strings. So in effect we can only set toplevelVar once.

Reply:
You could set it with a different priority, such as `mkDefault` to hack around that. The real solution is to fix the type to not merge strings.
Can be done later. It's internal.